src/main/java/com/theelixrlabs/healthcare/service/DoctorPatientAssignmentService.java

```diff
@@ -48,6 +48,34 @@ public DoctorPatientAssignmentDto assignDoctorToPatient(DoctorPatientAssignmentD
         return mapDoctorPatientAssignment(doctorPatientAssignmentDto);
     }
 
+    /**
+     * checks for doctor patient assignment and if present then unassigns doctor from patient
+     *
+     * @param doctorPatientAssignmentDto DTO object containing doctorId and patientId
+     * @throws CustomException if any exception occurs
+     */
+    public void unassignDoctorFromPatient(DoctorPatientAssignmentDto doctorPatientAssignmentDto) throws CustomException {
+        validateDoctorPatientAssignment(doctorPatientAssignmentDto);
+        UUID doctorId = UUID.fromString(doctorPatientAssignmentDto.getDoctorId());
```

Comment:
this can be moved to a private method convertStringToUuid() which will return the UUID value and also id validation can be moved from validateDoctorPatientAssignment()
by above change you can
call validateDoctorPatientAssignment after line 60 and their just check the doctor patient existance

Reply:
resolved